frontend/src/lib/components/UnitPicker/CustomUnitModal.tsx

```diff
@@ -0,0 +1,91 @@
+import { PureField } from 'lib/forms/Field'
+import { capitalizeFirstLetter } from 'lib/utils'
+import { LemonInput } from 'lib/components/LemonInput/LemonInput'
+import { HandleUnitChange } from 'lib/components/UnitPicker/UnitPicker'
+
+function chooseFormativeElementValue(
+    formativeElement: 'prefix' | 'postfix' | null,
+    filters: Partial<FilterType>
+): string {
+    if (formativeElement === 'prefix') {
+        return filters.aggregation_axis_prefix || ''
+    }
+
+    if (formativeElement === 'postfix') {
+        return filters.aggregation_axis_postfix || ''
+    }
+
+    return ''
+}
+
+export function CustomUnitModal({
+    isOpen,
+    onSave,
+    formativeElement,
+    filters,
+    onClose,
+    overlayRef,
+}: {
+    isOpen: boolean
+    onSave: (hx: HandleUnitChange) => void
+    formativeElement: 'prefix' | 'postfix' | null
+    filters: Partial<FilterType>
+    onClose: () => void
+    overlayRef: RefCallback<HTMLDivElement>
+}): JSX.Element | null {
+    const [localFormativeElementValue, setLocalFormativeElementValue] = useState<string>(
+        chooseFormativeElementValue(formativeElement, filters)
+    )
+    useEffect(() => {
+        setLocalFormativeElementValue(chooseFormativeElementValue(formativeElement, filters))
+    }, [formativeElement])
+
+    if (formativeElement === null) {
+        return null
+    }
+
+    return (
+        <LemonModal
+            overlayRef={overlayRef}
+            isOpen={isOpen}
+            onClose={onClose}
+            bringToFront={true}
+            title={`Custom ${formativeElement}`}
+            footer={
+                <>
+                    <div className="flex-1">
+                        <LemonButton type="secondary" data-attr="custom-prefix-cancel" onClick={onClose}>
+                            Cancel
+                        </LemonButton>
+                    </div>
+                    <LemonButton
+                        type="primary"
+                        onClick={() => onSave({ [formativeElement]: localFormativeElementValue })}
+                    >
+                        Apply
+                    </LemonButton>
+                </>
+            }
+        >
+            <PureField
+                label={`${capitalizeFirstLetter(formativeElement)}:`}
+                help={
+                    <>
+                        A {formativeElement} of <strong>{localFormativeElementValue || '$'}</strong> would mean `123.45`
+                        would be displayed{' '}
+                        <strong>
+                            {formativeElement === 'prefix' ? localFormativeElementValue || '$' : ''}123.45
+                            {formativeElement === 'postfix' ? localFormativeElementValue || '$' : ''}
+                        </strong>
+                    </>
+                }
+            >
+                <LemonInput value={localFormativeElementValue} onChange={setLocalFormativeElementValue} />
```

Comment:
Also noticed this doesn't have autofocus, which caught me off guard – I started typing right away